oslabs-beta/Kubiquity
# Add auto port forwarding for Prometheus

WHAT DOES THIS PR DO?

1. Adds auto port forwarding of the Prometheus service.
2. `export POD_NAME=$(kubectl get pods --all-namespaces -l "app=prometheus,component=server" -o jsonpath="{.items[0].metadata.name}") &&
  kubectl --namespace prometheus port-forward $POD_NAME 9090` no longer needs to be ran.

HOW TO TEST THIS PR:

1. Run: `gh pr checkout 28 && npm install`
2. Next you can either go into App.jsx and switch from the mock data to actual backend data by uncommenting the lines described in the TODO comments and then run `npm run dev` **OR** run `npm run backend` and send an HTTP request to `/metrics` endpoint.
3. Verify that data shows up from the request/in the app.
4. Verify that your terminal prints out `stdout: Handling connection for 9090`

ADDITIONAL NOTES:

You will need minikube running in the background. Also it is assumed you have a prometheus pod running.
